pallet/src/lib.rs

```diff
@@ -286,7 +287,8 @@ pub mod pallet {
                 )
             };
             if signer_count > 0 {
-                signature_handler.sign_script(&who, cheque_limit.into())?;
+                let lock_id = Self::multi_signer_lock_id(&who, &transaction_bc[..], &cheque_limit);
+                signature_handler.sign_script(&who, cheque_limit.into(), lock_id)?;
             }
```

Comment:
Self-note:
_We execute this part even for the single-signer scripts - let's discuss this on Slack - I have a general idea of how to avoid this - but that should be done in a separate PR_